### PR TITLE
 Add Feature Scaling to ML Pipeline

### PR DESCRIPTION
**Pull request by DevanshKhodaskar**
# 🚀 Enhancement: Improve Training Script with Pipeline and Feature Scaling

## 📌 Summary
Enhanced `train_and_export_model.py` with proper feature scaling and pipeline architecture for better model performance and reliability.

---

## 🔄 Changes
- ✅ Added **StandardScaler** for numerical features  
- ✅ Implemented **scikit-learn Pipeline** with `ColumnTransformer`  
- ✅ Replaced manual `pd.get_dummies()` with **OneHotEncoder**  
- ✅ Improved handling of **unknown categorical values** (`handle_unknown="ignore"`)  

---

## 🎯 Benefits
- Improved model performance through **feature normalization**  
- More **robust preprocessing pipeline** (no manual transformations required)  
- Better **production reliability** by exporting the complete pipeline instead of just the raw model  

---

## ⚠️ Breaking Change
The exported model artifact is now a **Pipeline object** instead of a raw `XGBClassifier`.  
This may require updates in `main.py` or any scripts consuming the model.  

---

## 🆚 Old vs New Approach
```python
# 🔴 Old Approach
# Manual one-hot encoding
df = pd.get_dummies(df, columns=categorical_cols, drop_first=True)

# No feature scaling
model = XGBClassifier(eval_metric='logloss')
```
# 🟢 New Approach
```python
# Structured preprocessing pipeline
preprocessor = ColumnTransformer(
    transformers=[
        ("num", StandardScaler(), numerical_cols),
        ("cat", OneHotEncoder(handle_unknown="ignore"), categorical_cols)
    ]
)

pipeline = Pipeline([
    ("preprocess", preprocessor),
    ("model", XGBClassifier(eval_metric="logloss", use_label_encoder=False))
])
```
📂 Files Modified

train_and_export_model.py → Complete pipeline rewrite

🔖 Type of Change

Enhancement

🏷️ Labels

ml-pipeline, feature-scaling

### CODE REVIEW

**Comment by DevanshKhodaskar** [comment]
Pelase assign me this under hackatoberfest
